File: app/src/main/java/com/zy/ppmusic/utils/PrintLog.java
package com.zy.ppmusic.utils;

import android.util.Log;

import com.zy.ppmusic.BuildConfig;

import java.util.Locale;

/**
 * @author ZhiTouPC
 * @date 2017/12/27
 */

public class PrintLog {
    private static String mClassName;
    private static String mMethodName;
    private static int mLineNumber;
    private static String mDetailInfo;


    private static void generateInfo(StackTraceElement[] elements) {
        mClassName = elements[1].getClassName();
        mMethodName = elements[1].getMethodName();
        mLineNumber = elements[1].getLineNumber();
        mDetailInfo = String.format(Locale.CHINA, "%s : %s : %s \n", mClassName, mMethodName, mLineNumber);
        System.out.println("--**--");
    }

    public static void print(Object msg) {
        generateInfo(new Throwable().getStackTrace());
        System.out.println(mDetailInfo + msg);
    }

    public static void e(String msg) {
        generateInfo(new Throwable().getStackTrace());
        Log.e(mClassName, buildLogMsg(msg));
    }

    public static void w(String msg) {
        generateInfo(new Throwable().getStackTrace());
        Log.w(mClassName, buildLogMsg(msg));
    }

    public static void i(String msg) {
        generateInfo(new Throwable().getStackTrace());
        Log.i(mClassName, buildLogMsg(msg));
    }

    public static void d(String msg) {
        generateInfo(new Throwable().getStackTrace());
        Log.d(mClassName, buildLogMsg(msg));
    }

    private static String buildLogMsg(String msg) {
        if (BuildConfig.DEBUG) {
            return String.format(Locale.CHINA, "%s -> %d \n %s", mMethodName, mLineNumber, msg == null?"empty":msg);
        } else {
            return "-------------------------------";
        }
    }
}

File: app/src/main/java/com/zy/ppmusic/utils/PrintLog.kt
package com.zy.ppmusic.utils

import android.util.Log

import com.zy.ppmusic.BuildConfig

import java.util.Locale

/**
 * @author ZhiTouPC
 * @date 2017/12/27
 */

object PrintLog {
    private var mClassName: String? = null
    private var mMethodName: String? = null
    private var mLineNumber: Int = 0
    private var mDetailInfo: String? = null

    private fun generateInfo(elements: Array<StackTraceElement>) {
        mClassName = elements[1].className
        mMethodName = elements[1].methodName
        mLineNumber = elements[1].lineNumber
        mDetailInfo = String.format(Locale.CHINA, "%s : %s : %s \n", mClassName, mMethodName, mLineNumber)
        println("--**--")
    }

    fun print(msg: Any) {
        generateInfo(Throwable().stackTrace)
        println(mDetailInfo!! + msg)
    }

    fun e(msg: String) {
        generateInfo(Throwable().stackTrace)
        Log.e(mClassName, buildLogMsg(msg))
    }

    fun w(msg: String) {
        generateInfo(Throwable().stackTrace)
        Log.w(mClassName, buildLogMsg(msg))
    }

    fun i(msg: String) {
        generateInfo(Throwable().stackTrace)
        Log.i(mClassName, buildLogMsg(msg))
    }

    fun d(msg: String) {
        generateInfo(Throwable().stackTrace)
        Log.d(mClassName, buildLogMsg(msg))
    }

    private fun buildLogMsg(msg: String?): String {
        return if (BuildConfig.DEBUG) {
            String.format(Locale.CHINA, "%s -> %d \n %s", mMethodName, mLineNumber, msg ?: "empty")
        } else {
            "-------------------------------"
        }
    }
}

fun loge(msg:String){
    PrintLog.e(msg)
}

fun logd(msg:String){
    PrintLog.d(msg)
}

fun logi(msg:String){
    PrintLog.i(msg)
}

fun logw(msg:String){
    PrintLog.w(msg)
}
